src/lib/generateDefinition.ts

```diff
@@ -4,30 +4,53 @@ import {getClusterAttributeValue} from './utils';
 import * as m from './modernExtend';
 import * as zh from 'zigbee-herdsman/dist';
 import {philipsLight} from './philips';
+import {Device} from 'zigbee-herdsman/dist/controller/model';
 
-interface GeneratedExtend {extend: ModernExtend, source: string, lib?: string}
+interface GeneratedExtend {extend?: ModernExtend, extendFn?: (a: object) => ModernExtend, args?: object, source: string, lib?: string}
```

Comment:
I would prefer it to have fields like `extendFn?: (a: T) => ModernExtend, args?: T`, but I am not sure if it is possible to achieve this.

Reply:
Why not?

Reply:
I am not strong at Typescript(that's one reason), but from other languages I know that if I would to add a type parameter here, then it would mean that in all uses of this interface there should be a type parameter specified.

For example, if `GeneratedExtend` would be `GeneratedExtend<T>`, then what this should look like: 
```ts
type ExtendGenerator = (endpoints?: Zh.Endpoint[]) => Promise<GeneratedExtend[]>;
```
? If to provide single type here then we will not be able to have extender functions that will accept different set of argument types.

Maybe the other way is to provide a class that has two methods like
```ts
class Provider {
    getModernExtend(): ModernExtend
    getSource(): string
}
```
and to have this be implemented by other type, that is generic?

What would you suggest?

Reply:
I tried to do it with the generic but failed to do so. For functions like `extenderOnOffLight` it's also quite complicated as args can be either `OnOffArgs` or `LightArgs`. Therefore the provider class sounds like the best solution

Reply:
Updated. Please check if solution is good.